### ISSUE TITLE
HTTP transport problem

### ISSUE THREAD

**Issue by isbalashov** [jaegertracing/jaeger-client-node#353]
## Software versions
For our tests we used:
* jaeger-client-node: v3.12.x - v3.14.4
* jaeger-collector: v1-9 - v1.11
## Requirement - what kind of business use case are you trying to solve?

Client library doesn't close sockets, thus with a big number of request parallyzing network communication leading to **EADDRNOTAVAIL** error

```
Failed to flush spans in reporter: error sending spans over HTTP: Error: connect EADDRNOTAVAIL 172.30.20.40:14268 - Local (10.110.2.33:0)
error sending spans over HTTP: Error: connect EADDRNOTAVAIL 172.30.20.40:14268 - Local (10.130.2.33:0)
Failed to flush spans in reporter: error sending spans over HTTP: Error: connect EADDRNOTAVAIL 172.30.20.40:14268 - Local (10.110.2.33:0)
error sending spans over HTTP: Error: connect EADDRNOTAVAIL 172.30.20.40:14268 - Local (10.130.2.33:0)
Failed to flush spans in reporter: error sending spans over HTTP: Error: connect EADDRNOTAVAIL 172.30.20.40:14268 - Local (10.130.2.33:0)
error sending spans over HTTP: Error: connect EADDRNOTAVAIL 172.30.20.40:14268 - Local (10.130.2.33:0)
```

## Problem - what in Jaeger blocks you from solving the requirement?

**HTTP transport problem**
When http used as main ransport, jaeger-client-node doesn't cleanup sockets,
with lots of  requests (e.g 100k) Linux faces socket shortage, number of opensockets increases up to max limit of 65k and that parallazyes network communication

![image](https://user-images.githubusercontent.com/29139808/55230329-913b4800-5230-11e9-95ba-eb4106072798.png)

## Proposal - what do you suggest to solve the problem or improve the existing situation?

To prevent socket shortage for entire hosts, max socket limit for http_sender could be introduced,
but this has a downside: sender quickly reaches limit and stops sending traces.

**Comment by yrsh** [comment]
Seems that is the problem with collector, outbound connections from client-lib are not closing properly. When I add req.abort() to request callback in flush method of httpSender the problem disappears. I can make PR with this as workaround, anyway it will be better then failing on prod.

**Comment by tiffon** [comment]
This should be fixed in [3.15.0](https://github.com/jaegertracing/jaeger-client-node/releases/tag/v3.15.0).

**Comment by tiffon** [comment]
This should be fixed in [3.15.0](https://github.com/jaegertracing/jaeger-client-node/releases/tag/v3.15.0).

**Comment by tiffon** [comment]
Fixed by #343.
